tests/components/satel_integra/test_config_flow.py

```diff
@@ -0,0 +1,497 @@
+"""Test the satel integra config flow."""
+
+from typing import Any
+from unittest.mock import AsyncMock
+
+import pytest
+
+from homeassistant.components.binary_sensor import BinarySensorDeviceClass
+from homeassistant.components.satel_integra.const import (
+    CONF_ARM_HOME_MODE,
+    CONF_DEVICE_PARTITIONS,
+    CONF_OUTPUT_NUMBER,
+    CONF_OUTPUTS,
+    CONF_PARTITION_NUMBER,
+    CONF_SWITCHABLE_OUTPUT_NUMBER,
+    CONF_SWITCHABLE_OUTPUTS,
+    CONF_ZONE_NUMBER,
+    CONF_ZONE_TYPE,
+    CONF_ZONES,
+    DEFAULT_PORT,
+    DOMAIN,
+    SUBENTRY_TYPE_OUTPUT,
+    SUBENTRY_TYPE_PARTITION,
+    SUBENTRY_TYPE_SWITCHABLE_OUTPUT,
+    SUBENTRY_TYPE_ZONE,
+)
+from homeassistant.config_entries import (
+    SOURCE_IMPORT,
+    SOURCE_RECONFIGURE,
+    SOURCE_USER,
+    ConfigSubentry,
+    ConfigSubentryData,
+)
+from homeassistant.const import CONF_CODE, CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+CONST_HOST = "192.168.0.2"
+CONST_PORT = 7095
+CONST_CODE = "1234"
+
+
+@pytest.mark.parametrize(
+    ("user_input", "entry_data", "entry_options"),
+    [
+        (
+            {CONF_HOST: CONST_HOST, CONF_PORT: CONST_PORT, CONF_CODE: CONST_CODE},
+            {CONF_HOST: CONST_HOST, CONF_PORT: CONST_PORT},
+            {CONF_CODE: CONST_CODE},
+        ),
+        (
+            {
+                CONF_HOST: CONST_HOST,
+            },
+            {CONF_HOST: CONST_HOST, CONF_PORT: DEFAULT_PORT},
+            {CONF_CODE: None},
+        ),
+    ],
+)
+async def test_setup_flow(
+    hass: HomeAssistant,
+    mock_satel: AsyncMock,
+    mock_setup_entry: AsyncMock,
+    user_input: dict[str, Any],
+    entry_data: dict[str, Any],
+    entry_options: dict[str, Any],
+) -> None:
+    """Test the setup flow."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input,
+    )
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == CONST_HOST
+    assert result["data"] == entry_data
+    assert result["options"] == entry_options
+
+    assert len(mock_setup_entry.mock_calls) == 1
```

Comment:
One thing to keep in mind, we should not allow multiple entries for the same device to be set up. So even though `single_config_entry` is true, if we have something unique that we can use for the unique id in the future, it would be nice that we can already set that during import

Reply:
Yea, the only thing we can use is the unique ID of the config flow itself, no other identifier (that are available from the alarm system itself) are valid. Do we need to set the unique ID using the ID of the config flow manually?